src/hooks/useFetch.js

```diff
@@ -1,20 +1,27 @@
 import { useEffect, useState } from 'react';
-import { STORAGE_TOKEN } from '../helper/variables';
+import { API_EDPOIONTS, STORAGE_TOKEN } from '../helper/variables';
 
-function useFetch(endpoint, searchQuery) {
+function useFetch(endpoint, searchQuery, id) {
   const [isLoading, setIsLoading] = useState(false);
   const [error, setError] = useState(null);
   const [data, setData] = useState([]);
+  const URL = id
+    ? `http://localhost:3333/pfa.foreca.com/api/v1/${endpoint}/${id}`
+    : `http://localhost:3333/pfa.foreca.com/api/v1/location/${endpoint}/${searchQuery}`;
```

Comment:
this is not a very extendable way to do it. Think about how you can just pass some params and directly update the url